contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,367 @@
+The feature will be enabled by default.
 
-Examples in the next section.
+As a result, we will recommend that users drain their nodes prior
+to upgrade of the `kubelet`.  If users do not drain their nodes, the
+`kubelet` will act as follows:
 
-### Proposed cgroup hierarchy:
+1. If a pod has a `RestartPolicy=Never`, then mark the pod
+as `Failed` and terminate its workload.
+1. All other pods that are not parented by a pod-level cgroup
+will be restarted.
 
-For the initial implementation we will only support limits for cpu and memory resources.
+The `cgroups-per-qos` flag will be enabled by default, but user's
+may choose to opt-out.  We may deprecate this opt-out mechanism
+in Kubernetes 1.7, and remove the flag entirely in Kubernetes 1.8.
 
-#### QoS classes
+#### Risk Assessment
 
-A pod can belong to one of the following 3 QoS classes: Guaranteed, Burstable, and BestEffort, in decreasing order of priority.
+The impact of the unified cgroup hierarchy is restricted to the `kubelet`.
 
-#### Guaranteed
+Potential issues:
 
-`G` pods will be placed at the `$Root` cgroup by default. `$Root` is the system root i.e. "/" by default and if `--cgroup-root` flag is used then we use the specified cgroup-root as the `$Root`. To ensure Kubelet's idempotent behaviour we follow a pod cgroup naming format which is opaque and deterministic. Say we have a pod with UID: `5f9b19c9-3a30-11e6-8eea-28d2444e470d` the pod cgroup PodUID would be named: `pod-5f9b19c93a3011e6-8eea28d2444e470d`.
+1. Bugs
+1. Performance and/or reliability issues for `BestEffort` pods.  This is
+most likely to appear on E2E test runs that mix/match pods across different
+QoS tiers.
+1. User misconfiguration; most notably the `--cgroup-driver` needs to match
+the expected behavior of the container runtime.  We provide clear errors
+in `kubelet` logs for container runtimes that we include in tree.
 
+#### Proposed Timeline
 
-__Note__: The cgroup-root flag would allow the user to configure the root of the QoS cgroup hierarchy. Hence cgroup-root would be redefined as the root of QoS cgroup hierarchy and not containers.
+* 01/31/2017 - Discuss the rollout plan in sig-node meeting
+* 02/14/2017 - Flip the switch to enable pod level cgroups by default
+ * enable existing experimental behavior by default
+* 02/21/2017 - Assess impacts based on enablement
+* 02/27/2017 - Kubernetes Feature complete (i.e. code freeze)
+ * opt-in behavior surrounding the feature (`qos-memory-overcommit` support) completed.
+* 03/01/2017 - Send an announcement to kubernetes-dev@ about the rollout and potential impact
+* 03/22/2017 - Kubernetes 1.6 release
+* TBD (1.7?) - Deprecate the old implementation
```

Comment:
i meant this to read look at deprecating the ability to have pod level cgroups off.  will clarify text.